### PR TITLE
Add a note on PodsReady timeout requeuing

### LINKED ISSUE #2009: [WaitForPodsReady] The default configuration for `requeuingStrategy` is impractical
**What happened**:

Consider the following configuration:

```yaml
waitForPodsReady:
  enable: true
  timeout: 1m
  blockAdmission: false
  requeuingStrategy:
    timestamp: Creation
    backoffLimitCount: 4
 ```
And let's say we have a workload which have unschedulable pods (for whatever reason).

If a job was timed-out after 1min (or 10min) of waiting, then requeue after 1s is not relevant to end-users.
it takes around 10 requeues for the backoff delay to be relevant (about 1min).

**What you expected to happen**:

The default base backoff should be larger than 10s to make it relevant to end-users.
Make the default and potentially (TBD) configurable.

If we agree on the default of 10s we could back-port it to 0.6 branch.

**How to reproduce it (as minimally and precisely as possible)**:

Example global configuration:
```yaml
waitForPodsReady:
  enable: true
  timeout: 1m
  blockAdmission: false
  requeuingStrategy:
    timestamp: Creation
    backoffLimitCount: 4
 ```
 Example cluster configuration:
 ```yaml
apiVersion: kueue.x-k8s.io/v1beta1
kind: ResourceFlavor
metadata:
  name: "default-flavor"
---
apiVersion: kueue.x-k8s.io/v1beta1
kind: ClusterQueue
metadata:
  name: "cluster-queue"
spec:
  namespaceSelector: {}
  resourceGroups:
  - coveredResources: ["cpu", "memory"]
    flavors:
    - name: "default-flavor"
      resources:
      - name: "cpu"
        nominalQuota: 200
      - name: "memory"
        nominalQuota: 36Gi
---
apiVersion: kueue.x-k8s.io/v1beta1
kind: LocalQueue
metadata:
  namespace: "default"
  name: "user-queue"
spec:
  clusterQueue: "cluster-queue"
 ```
 Example Job:
```yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: sample-job
  namespace: default
  labels:
    kueue.x-k8s.io/queue-name: user-queue
spec:
  parallelism: 1
  completions: 1
  suspend: true
  template:
    spec:
      containers:
      - name: dummy-job
        image: gcr.io/k8s-staging-perf-tests/sleep:v0.0.3
        args: ["1800s"]
        resources:
          requests:
            cpu: "50"
            memory: "200Mi"
      restartPolicy: Never
```

Watching the events we can see:
```sh
> kubectl get workloads -w --output-watch-events       
EVENT      NAME                   QUEUE        ADMITTED BY   AGE
ADDED      job-sample-job-b1ca7   user-queue                 13m
MODIFIED   job-sample-job-b1ca7   user-queue                 13m
DELETED    job-sample-job-b1ca7   user-queue                 13m
ADDED      job-sample-job-08913   user-queue                 0s
MODIFIED   job-sample-job-08913   user-queue                 0s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   0s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   60s
MODIFIED   job-sample-job-08913   user-queue                   61s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   61s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   2m1s
MODIFIED   job-sample-job-08913   user-queue                   2m2s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   2m2s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   3m2s
MODIFIED   job-sample-job-08913   user-queue                   3m3s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   3m3s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   4m3s
MODIFIED   job-sample-job-08913   user-queue                   4m4s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   4m5s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   5m5s
MODIFIED   job-sample-job-08913   user-queue                   5m6s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   5m8s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   6m8s
MODIFIED   job-sample-job-08913   user-queue                   6m9s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   6m13s
MODIFIED   job-sample-job-08913   user-queue   cluster-queue   7m13s
MODIFIED   job-sample-job-08913   user-queue                   7m14s
```

**Anything else we need to know?**:

The algorithm is described [here](https://github.com/kubernetes-sigs/kueue/tree/main/keps/1282-pods-ready-requeue-strategy#exponential-backoff-mechanism).

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:

Fixes #2009

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/content/en/docs/tasks/manage/setup_sequential_admission.md [hunk 1]
```diff
@@ -93,6 +94,12 @@ If you don't specify any value for `backoffLimitCount`,
 a Workload is repeatedly and endlessly re-queued to the queue based on the `timestamp`.
 Once the number of re-queues reaches the limit, Kueue [deactivates the Workload](/docs/concepts/workload/#active).
 
+The time to reque a workload after each consecutive timeout is increased
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/manage/setup_sequential_admission.md:97]:
```suggestion
The time to re-queue a workload after each consecutive timeout is increased
```

File: site/content/en/docs/tasks/manage/setup_sequential_admission.md [hunk 2]
```diff
@@ -93,6 +94,12 @@ If you don't specify any value for `backoffLimitCount`,
...
+exponentially, with the exponent of 2. The first delay is determined by the
+`backoffBaseSeconds` parameter (defaulting to 10). There is additionally a small
+random jitter added to the time to requeue. So, after the consecutive timeouts
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/manage/setup_sequential_admission.md:100]:
```suggestion
random jitter added to the time to re-queue. So, after the consecutive timeouts
```

Based on the other places.

File: site/content/en/docs/tasks/manage/setup_sequential_admission.md [hunk 3]
```diff
@@ -93,6 +94,12 @@ If you don't specify any value for `backoffLimitCount`,
...
+the evicted workload is requeued after approximately `10, 20, 40, ...` seconds.
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/manage/setup_sequential_admission.md:101]:
```suggestion
the evicted workload is re-queued after approximately `10, 20, 40, ...` seconds.
```

File: site/content/en/docs/tasks/manage/setup_sequential_admission.md [hunk 4]
```diff
@@ -93,6 +94,12 @@ If you don't specify any value for `backoffLimitCount`,
 a Workload is repeatedly and endlessly re-queued to the queue based on the `timestamp`.
 Once the number of re-queues reaches the limit, Kueue [deactivates the Workload](/docs/concepts/workload/#active).
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/manage/setup_sequential_admission.md:96]:
```suggestion

{{% alert title="Warning" color="warning" %}}
_The `backoffBaseSeconds` is available in Kueue v0.7.0 and later_
{{% /alert %}}
```

File: site/content/en/docs/tasks/manage/setup_sequential_admission.md [hunk 5]
```diff
@@ -93,6 +94,15 @@ If you don't specify any value for `backoffLimitCount`,
...
 
+{{% alert title="Warning" color="warning" %}}
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply site/content/en/docs/tasks/manage/setup_sequential_admission.md:97]: Note is good enough.

File: site/content/en/docs/tasks/manage/setup_sequential_admission.md [hunk 6]
```diff
@@ -93,6 +94,15 @@ If you don't specify any value for `backoffLimitCount`,
...
+_The `backoffBaseSeconds` is available in Kueue v0.7.0 and later_
+{{% /alert %}}
+The time to re-queue a workload after each consecutive timeout is increased
+exponentially, with the exponent of 2. The first delay is determined by the
+`backoffBaseSeconds` parameter (defaulting to 10). There is additionally a small
+random jitter added to the time to re-queue. So, after the consecutive timeouts
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/manage/setup_sequential_admission.md:103]: Remove the jitter bit. It's TMI for users.

**Review by alculquicondor — COMMENTED** [review]
/approve

File: site/content/en/docs/tasks/manage/setup_sequential_admission.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (tenzen-y) [reply site/content/en/docs/tasks/manage/setup_sequential_admission.md:97]: I'm ok with `Note`. Thank you for the suggestions!

File: site/content/en/docs/tasks/manage/setup_sequential_admission.md (hunk 6 above)
Comment (mimowo) [reply site/content/en/docs/tasks/manage/setup_sequential_admission.md]: Done

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm

**Pull request merged**